packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -628,12 +632,15 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
                     // TODO(bkonyi): this won't handle text zoom correctly,
                     //  https://github.com/flutter/flutter/issues/48522
                     height: 56.0,
-                    child: SingleChildScrollView(
-                      dragStartBehavior: widget.dragStartBehavior,
-                      scrollDirection: Axis.horizontal,
-                      reverse: true,
-                      child: Row(
-                        children: footerWidgets,
+                    child: Align(
+                      alignment: widget.footerRowAlignment,
+                      child: SingleChildScrollView(
```

Comment:
This alignment wrapped around a scrollable like this could have cases where it does not work as expected. If the footer widgets are longer that the width of the viewport, how is the align applied?

This PR will need tests before we can land it, so can you add a test for that scenario along with the scenarios you pictured in the description?